### PR TITLE
Add loading icon and disable continue until free extensions are loaded

### LINKED ISSUE #7321: OBW business step extensions are not installed if user continued before plugin list is loaded
### Describe the bug
Recently we've added RemoteFreeExtension feature which loads the business step extensions from a remote source. It seems that if a user clicks on "Continue" quick enough before the list of plugins are loaded, it will not install any of the suggested plugins and skips the whole step without performing any action. 

### To Reproduce
1. On a fresh site, load the OBW.
2. In the business detail step, go to "free features" tab and click on "Continue" before the plugins are loaded.
3. Observe the step proceeds without any plugin installation.

### Expected behavior
I think the best course of action is not to let users proceed before the plugins are loaded.

### PR DESCRIPTION
**Pull request by louwie17**
Fixes #7321 

Make sure the free extensions are loaded before a user can hit continue.

### Screenshots

Kapture 2021-12-01 at 10 36 39

### Detailed test instructions:

- Delete the `woocommerce_admin_remote_free_extensions_specs` transient so the free extensions takes a little longer
- Start the Onboarding flow and continue to the Business details step
- Fill out the dropdown's and click continue so it goes to the free extension tab
- Notice how the `continue` button is disabled until the free extensions are loaded and there is a loading icon present

### CODE REVIEW

**Review by joshuatf — APPROVED** [review]
LGTM pending @elizaan36's approval of the spinner.  Nice work!

Might also be worth hiding or disabling the expansion button (chevron down) while those are loading as I know I've been able to toggle that in the past prior to extensions being fully loaded.

**Comment by louwie17** [comment]
> Might also be worth hiding or disabling the expansion button (chevron down) while those are loading as I know I've been able to toggle that in the past prior to extensions being fully loaded.

Thanks for the review Josh, and good suggestion, I added this logic incase you wanted to check it again. I only show the chevron if there are free extensions to show.

**Comment by elizaan36** [comment]
> @elizaan36 I added this loading icon before the checkbox shows up, does this look ok? I can remove it as well if you prefer just disabling the continue button.

The loading icon is a little noisy with that placement on this screen. I just had a look at the original issue and I think disabling the CTA with a shimmer load indication until the extensions are loaded will solve the problem. A shimmer load on the disabled button would give context to users that something is happening in the background for the second or two that they can't move forward.

I'd rather use that visual indication of loading to minimize distractions and be consistent with other screens.

**Comment by louwie17** [comment]
> The spinner is a little noisy with that placement on this screen. I just had a look at the original issue and I think disabling the CTA with a shimmer load indication until the extensions are loaded will solve the problem. A shimmer load on the disabled button would give context to users that something is happening in the background for the second or two that they can't move forward.
> I'd rather use that visual indication of loading to minimize distractions and be consistent with other screens.

Thanks @elizaan36, that sounds good to me.

**Comment by louwie17** [comment]
@joshuatf I updated this to reflect Elizabeth's suggestions, if you are able to give it another look, no worries if not.

**Review by octaedro — APPROVED** [review]
Good job @louwie17. This is testing well here and the code looks good 🚀

**Pull request merged**
